apoc.get.rels() rejects some ID arrays

## Expected Behavior (Mandatory)

`apoc.get.rels()` should accept all relationship ID arrays.

## Actual Behavior (Mandatory)

`apoc.get.rels()` fails with this error for some relationship ID arrays:

    Failed to invoke procedure `apoc.get.rels`: Caused by: java.lang.RuntimeException: Can't convert class [J to a Relationship

## How to Reproduce the Problem

### Steps

  1. Execute this query to see the error:

```
MERGE (g:Foo {id: 1})
MERGE (h:Foo {id: 2})
MERGE (g)-[r:BAR]->(h)
MERGE (t:Temp)
SET t.ids = [] + ID(r)
WITH t
CALL apoc.get.rels(t.ids) YIELD rel
RETURN rel
```
2. Replace `apoc.get.rels(t.ids)` with `apoc.get.rels([id IN t.ids | id])` to see the expected behavior.

## Specifications (Mandatory)

Currently used versions

### Versions

  - OS: MacOS 10.15.6
  - Neo4j: 4.1.1
  - Neo4j-Apoc: 4.1.0.2

--- a/core/src/main/java/apoc/convert/Convert.java
+++ b/core/src/main/java/apoc/convert/Convert.java
@@ -13,7 +13,14 @@ import org.neo4j.procedure.Description;
 import org.neo4j.procedure.Name;
 import org.neo4j.procedure.UserFunction;
 
-import java.util.*;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
@@ -71,15 +78,25 @@ public class Convert {
     }
 
     @SuppressWarnings("unchecked")
-    private List convertToList(Object list) {
+    public static List convertToList(Object list) {
         if (list == null) return null;
         else if (list instanceof List) return (List) list;
         else if (list instanceof Collection) return new ArrayList((Collection)list);
         else if (list instanceof Iterable) return Iterators.addToCollection(((Iterable)list).iterator(),(List)new ArrayList<>(100));
         else if (list instanceof Iterator) return Iterators.addToCollection((Iterator)list,(List)new ArrayList<>(100));
-        else if (list.getClass().isArray() && !list.getClass().getComponentType().isPrimitive()) {
-            List result = new ArrayList<>(100);
-            Collections.addAll(result, ((Object[]) list));
+        else if (list.getClass().isArray()) {
+            final Object[] objectArray;
+            if (list.getClass().getComponentType().isPrimitive()) {
+                int length = Array.getLength(list);
+                objectArray = new Object[length];
+                for (int i = 0; i < length; i++) {
+                    objectArray[i] = Array.get(list, i);
+                }
+            } else {
+                objectArray = (Object[]) list;
+            }
+            List result = new ArrayList<>(objectArray.length);
+            Collections.addAll(result, objectArray);
             return result;
         }
         return null;

--- a/core/src/main/java/apoc/util/Util.java
+++ b/core/src/main/java/apoc/util/Util.java
@@ -1,6 +1,7 @@
 package apoc.util;
 
 import apoc.Pools;
+import apoc.convert.Convert;
 import apoc.export.util.CountingInputStream;
 import org.apache.commons.io.IOUtils;
 import org.apache.commons.lang.StringUtils;
@@ -146,14 +147,7 @@ public class Util {
     }
 
     public static Stream<Object> stream(Object values) {
-        if (values == null) return Stream.empty();
-        if (values instanceof Collection) return ((Collection)values).stream();
-        if (values instanceof Object[]) return Stream.of(((Object[])values));
-        if (values instanceof Iterable) {
-            Spliterator<Object> spliterator = ((Iterable) values).spliterator();
-            return StreamSupport.stream(spliterator,false);
-        }
-        return Stream.of(values);
+        return Convert.convertToList(values).stream();
     }
 
     public static Stream<Node> nodeStream(Transaction tx, Object ids) {

--- a/full/src/test/java/apoc/get/GetProceduresTest.java
+++ b/full/src/test/java/apoc/get/GetProceduresTest.java
@@ -6,13 +6,16 @@ import org.junit.Rule;
 import org.junit.Test;
 import org.neo4j.graphdb.Node;
 import org.neo4j.graphdb.Relationship;
+import org.neo4j.graphdb.ResourceIterator;
 import org.neo4j.test.rule.DbmsRule;
 import org.neo4j.test.rule.ImpermanentDbmsRule;
 
+import java.util.Collections;
 import java.util.List;
 
 import static apoc.util.MapUtil.map;
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
 
 /**
  * @author mh
@@ -45,6 +48,24 @@ public class GetProceduresTest {
             assertEquals(true, ids.contains(((Relationship) r.next().get("rel")).getId()));
             assertEquals(true, ids.contains(((Relationship) r.next().get("rel")).getId()));
             assertEquals(true, ids.contains(((Relationship) r.next().get("rel")).getId()));
+        });
+    }
+
+    @Test
+    public void testArrayOfIds() {
+        String query = "MERGE (g:Foo {id: 1})\n" +
+                "MERGE (h:Foo {id: 2})\n" +
+                "MERGE (g)-[r:BAR]->(h)\n" +
+                "MERGE (t:Temp)\n" +
+                "SET t.ids = [] + ID(r)\n" +
+                "WITH t\n" +
+                "CALL apoc.get.rels(t.ids) YIELD rel\n" +
+                "RETURN rel";
+        TestUtil.testResult(db, query, Collections.emptyMap(), r -> {
+            final ResourceIterator<Relationship> relIT = r.columnAs("rel");
+            final Relationship rel = relIT.next();
+            assertEquals("BAR", rel.getType().name());
+            assertFalse(r.hasNext());
         });
     }
 }

